signer/signer_trust.go

```diff
@@ -1,20 +1,30 @@
 package signer
 
 import (
+	"errors"
+	"fmt"
 	"net"
+	"time"
 
 	"github.com/Sirupsen/logrus"
 	pb "github.com/docker/notary/proto"
 	"github.com/endophage/gotuf/data"
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials"
 )
 
+// The only thing needed from grpc.ClientConn is it's state.
+type checkableConnectionState interface {
+	State() grpc.ConnectivityState
+}
+
 // NotarySigner implements a RPC based Trust service that calls the Notary-signer Service
 type NotarySigner struct {
```

Comment:
As per discussion with @endophage, unexporting this does not break any tests, but it does break lint because `NewNotarySigner` returns an object that is unexported, so we are leaving this as is for now.